### PR TITLE
Fix some problems with string handling in src/python/WMCore/Services/Requests.py

### LINKED ISSUE #10507: [py2py3] problem with strings and base64 when contacting couchdb
**Impact of the bug**
All WMCore

**Describe the bug**
When working on #10502, after providing the fix in #10504 I get this error [1]. Alan already pointed me to [his approach](https://github.com/amaltaro/couchapp/commit/2d7084d6716c665bb3c17736eda1784e51210a2e) when dealing with a similar issue.

**How to reproduce it**

Run the following unittest
```bash
python3 setup.py test --buildBotMode=true --reallyDeleteMyDatabaseAfterEveryTest=true --testCertainPath=test/python/WMCore_t/WMBS_t/JobSplitting_t/EventBased_t.py
```
In a py3 environment with #10504 applied

**Expected behavior**
We should not get such a failure

**Additional context and error message**

[1]

```plaintext
======================================================================
ERROR: _testACDCProduction_v1_
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/dmwm/unittestdeploy/wmagent/1.4.9.pre3/sw/slc7_amd64_gcc630/external/python3/3.8.2-comp/lib/python3.8/base64.py", line 510, in _input_type_check
    m = memoryview(s)
TypeError: memoryview: a bytes-like object is required, not 'str'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/dmwm/wmcore_unittest/WMCore/test/python/WMCore_t/WMBS_t/JobSplitting_t/EventBased_t.py", line 46, in setUp
    self.testInit.setupCouch(self.couchDBName, "GroupUser", "ACDC")
  File "/home/dmwm/wmcore_unittest/WMCore/src/python/WMQuality/TestInitCouchApp.py", line 103, in setupCouch
    self.couch = CouchAppTestHarness(dbName)
  File "/home/dmwm/wmcore_unittest/WMCore/src/python/WMQuality/TestInitCouchApp.py", line 44, in __init__
    self.couchServer = CouchServer(self.couchUrl)
  File "/home/dmwm/wmcore_unittest/WMCore/src/python/WMCore/Database/CMSCouch.py", line 946, in __init__
    CouchDBRequests.__init__(self, url=dburl, usePYCurl=usePYCurl, ckey=ckey, cert=cert, capath=capath)
  File "/home/dmwm/wmcore_unittest/WMCore/src/python/WMCore/Database/CMSCouch.py", line 96, in __init__
    JSONRequests.__init__(self, url,
  File "/home/dmwm/wmcore_unittest/WMCore/src/python/WMCore/Services/Requests.py", line 542, in __init__
    Requests.__init__(self, url, idict)
  File "/home/dmwm/wmcore_unittest/WMCore/src/python/Utils/PortForward.py", line 69, in portMangle
    return callFunc(callObj, url, *args, **kwargs)
  File "/home/dmwm/wmcore_unittest/WMCore/src/python/WMCore/Services/Requests.py", line 92, in __init__
    self.addBasicAuth(urlComponent['username'], urlComponent['password'])
  File "/home/dmwm/wmcore_unittest/WMCore/src/python/WMCore/Services/Requests.py", line 413, in addBasicAuth
    auth_string = "Basic %s" % base64.encodestring('%s:%s' % (
  File "/home/dmwm/unittestdeploy/wmagent/1.4.9.pre3/sw/slc7_amd64_gcc630/external/python3/3.8.2-comp/lib/python3.8/base64.py", line 540, in encodestring
    return encodebytes(s)
  File "/home/dmwm/unittestdeploy/wmagent/1.4.9.pre3/sw/slc7_amd64_gcc630/external/python3/3.8.2-comp/lib/python3.8/base64.py", line 527, in encodebytes
    _input_type_check(s)
  File "/home/dmwm/unittestdeploy/wmagent/1.4.9.pre3/sw/slc7_amd64_gcc630/external/python3/3.8.2-comp/lib/python3.8/base64.py", line 513, in _input_type_check
    raise TypeError(msg) from err
TypeError: expected bytes-like object, not str
```

### PR DESCRIPTION
**Pull request by mapellidario**
Fixes #10507 

#### Status
Ready

#### Description
make string handling in `src/python/WMCore/Services/Requests.py` compatible with py3

#### Is it backward compatible (if not, which system it affects?)
yes

#### Related PRs
This PR needs to be rebased on
- [x] #10504 (mapellidario - 10502-fix )
- [x] #10501 (mapellidario - 10333-fix)

#### External dependencies / deployment changes
nope

### CODE REVIEW

**Comment by mapellidario** [comment]
This PR fixes the problem described at the issue #10507. 

The py3 unittest report says that we have 15 new failures, but actually those are unittests that were blocked by the error in #10507 and now fail at assertion time. I would prefer to merge this and open a new issue. @amaltaro what do you think?

**Comment by mapellidario** [comment]
I created #10521 to track the new issue

File: src/python/WMCore/Services/Requests.py [hunk 1]
```diff
@@ -410,8 +411,16 @@ def _getURLOpener(self):
 
     def addBasicAuth(self, username, password):
         """Add basic auth headers to request"""
-        auth_string = "Basic %s" % base64.encodestring('%s:%s' % (
+        ## TODO: base64.encodestring is deprecated
+        # https://docs.python.org/3.8/library/base64.html#base64.encodestring
+        # change to base64.encodebytes after we drop python2
+        username = encodeUnicodeToBytes(username)
+        password = encodeUnicodeToBytes(password)
+        encodedauth = base64.encodestring(b'%s:%s' % (
```
Comment (amaltaro) [inline src/python/WMCore/Services/Requests.py:419]: Dario, here you lost me. A few lines above, you say this `encodestring` has been deprecated in python3. Shouldn't we then either remove it or only execute this code for PY2 environment?

**Comment by amaltaro** [comment]
test this please

File: src/python/WMCore/Services/Requests.py (hunk 1 above)
Comment (mapellidario) [reply src/python/WMCore/Services/Requests.py:419]: Well, they say that `base64.encodedstring` is a deprecated alias for `base64.encodedbytes`, but it is still available as an alias in python3.8. I would keep it like this at the moment, then when we are over the migration we can move to `base64.encodedbytes` (which is not available in py2). Would you rather have an `if PY3` here to select already now the correct alias based on the python runtime version?

Comment (amaltaro) [reply src/python/WMCore/Services/Requests.py:419]: Your plan sounds good to me. If it works in python3.8, then we are good for the moment.

**Review by amaltaro — APPROVED** [review]
LGTM! Thanks Dario.

File: src/python/WMCore/Services/Requests.py (hunk 1 above)
Comment (mapellidario) [reply src/python/WMCore/Services/Requests.py:419]: Good, perfect, thank you!

**Pull request merged**